### DISCUSSION TITLE
Odd behavior in app mode parsing

### DISCUSSION THREAD

**Discussion by Mepper65** [maoserr/epublifier#388] [Q&A] [answered]
I'm working with this URL / book:
https://blossomtranslation.com/chapter/tistc-vol-1-chapter-01/
With a next selector of
._next:nth-child(2)
and a title selector of
.chapter__headline:nth-child(1) > h1

In the browser console I can run
document.getElementsByClassName("chapter__headline").item(0)
and verify that the expected element is selected.

Here is my screen:
![EpubEx03](https://github.com/user-attachments/assets/d6db24f9-b021-4213-bc42-44edfafc6e46)

And here is the log at the very end:
No automatic parser available for: blossomtranslation.com
Parsed simple chapter
Unable to find title element
Parsed simple chapter
Unable to find title element
Parsed simple chapter
Unable to find title element
Parsed simple chapter
Unable to find title element
Parsed simple chapter
Unable to find title element

The odd behavior is that when I click the add page button the app parses the current page and then loads the next page in the browser tab (as expected).  However, when the next page is loaded the app window disappears and it seems that the app stops/pauses (I assume the browser is controlling this behavior).  If I hit the browser back button to navigate back to the first page then the app window reappears and processing continues, which results in re-parsing the first page and then loading the second page again & the app window disappearing again.  If I go through this for the full page count (default 5) then at the end I'm back at the first page and if I save the epub I get the first chapter 5x (as expected given what the app shows).

Note that I specified a 5s per parse wait.  If I leave it at the default of 0.5s then the behavior looks like it is parsing the first page 5x and then loading the second page.  I used the longer wait value to try to see what was going on.  In Firefox the behavior is slightly different in that the per parse wait only happens when I am on the first page, meaning I can let the browser sit on the second page for any amount of time and then when I hit the back button the app reappears and sits for 5s before loading the second page again.  On Chrome the app seems to load the second page almost immediately.

I can still manage to create an epub by parsing a single page at a time and saving & loading cached data, but it is extremely tedious because the next & title selectors are not included in the cached data.  In other words it is possible, but not fun.

By the way, is there any easy way of testing the title & next selectors in the browser before running the app?  They appear to be CSS selectors.  My approach has been to look at the page source and do searches on elements & class names to figure out what should be reasonable.

**Comment by Mepper65** [comment]
I figured out one thing...

The search buttons next to the Next & Title selectors don't work on the current version of Firefox.  Since they never worked for me on Firefox I didn't think of trying them on Chrome.

When I tried them on Chrome for this website they came up with ".chapter__footer ._navigation" for the Next selector and ".chapter__title" for the Title selector.  The Title selector value is good.  The Next selector isn't good after the fist page.

I changed the Next selector to ".chapter__footer ._next" and retried the whole Add pages from above.  I had exactly the same results.

**Comment by maoserr** [comment, marked as answer]
that page is not an app, it's just a normal web page with TOC, I was able to parse the first 2 chapters just fine with this:
![image](https://github.com/user-attachments/assets/0f575f7d-5d10-41d2-b208-ef414419f563)

**Comment by Mepper65** [comment]
How do you quickly determine if a page should be consider a page or an app?  I've had Epublifier work & fail on different pages that look basically the same to me.  Do I just need to try everything both ways?  And in this particular case the pages all have next links so why doesn't treating it as an app work?

**Comment by maoserr** [comment]
Apps won't work if you disable javascript. If you're savvy with dev console, you should also see triggering next link triggers a js action for apps but doesn't refresh the dom.
